main.go

```diff
@@ -102,308 +124,285 @@ func exportEnvironmentWithEnvman(keyStr, valueStr string) error {
+	}
+	if err := diffFile.Close(); err != nil {
+		return "", err
+	}
 
-				return fetchShallowErr
-			}); err != nil {
-				log.Errorf("Failed, error: %s", err)
-				os.Exit(1)
-			}
+	return diffFile.Name(), nil
+}
 
-			if err := git.Checkout(); err != nil {
-				log.Errorf("Failed, error: %s", err)
-				os.Exit(1)
-			}
+func runWithRetry(f func() *command.Model) error {
+	return retry.Times(retryCount).Wait(waitTime).Try(func(attempt uint) error {
+		if attempt > 0 {
+			log.Warnf("Retrying...")
 		}
 
-		if git.ShouldMergePullRequest() {
-			if err := retry.Times(3).Wait(waitTime).Try(func(attempt uint) error {
-				if attempt > 0 {
-					log.Warnf("Retrying...")
-				}
+		err := run(f())
+		if err != nil {
+			log.Warnf("Attempt %d failed:", attempt+1)
+			fmt.Println(err.Error())
+		}
 
-				// the merge will be run max 4 times
-				// if 2. run fails, try to run normal merge instead of using the diff file
-				// if diff file does not exist, this flag has no effect to the merge
-				allowApplyDiffFile := (attempt < 2)
-				gitMergeErr := git.MergePullRequest(allowApplyDiffFile)
-				if gitMergeErr != nil {
-					log.Warnf("Attempt %d failed:", attempt+1)
-					fmt.Println(gitMergeErr.Error())
-				}
+		return err
+	})
+}
 
-				return gitMergeErr
-			}); err != nil {
-				log.Errorf("Failed, error: %s", err)
-				os.Exit(1)
-			}
-		}
+func printLog(format, env string) error {
+	l, err := runWithOutput(Git.Log(format))
+	if err != nil {
+		return err
+	}
 
-		if configs.UpdateSubmodules != "no" {
-			if err := retry.Times(retryCount).Wait(waitTime).Try(func(attempt uint) error {
-				if attempt > 0 {
-					log.Warnf("Retrying...")
-				}
+	log.Printf("=> %s\n   value: %s\n", env, l)
+	if err := exportEnvironmentWithEnvman(env, l); err != nil {
+		return fmt.Errorf("envman export failed, error: %v", err)
+	}
+	return nil
+}
 
-				submoduleErr := git.SubmoduleUpdate()
-				if submoduleErr != nil {
-					log.Warnf("Attempt %d failed:", attempt+1)
-					fmt.Println(submoduleErr.Error())
-				}
+func main() {
+	originPresent, err := isOriginPresent(configs.CloneIntoDir, configs.RepositoryURL)
+	if err != nil {
+		fail("Can't check if origin is presented, error: %v", err)
+	}
 
-				return submoduleErr
-			}); err != nil {
-				log.Errorf("Failed, error: %s", err)
-				os.Exit(1)
-			}
+	if originPresent && configs.ResetRepository == "yes" {
+		if err := resetRepo(); err != nil {
+			fail("Can't reset repository, error: %v", err)
 		}
+	}
 
-		log.Infof("Exporting git logs")
+	// Create directory if not exist
+	if err := os.MkdirAll(configs.CloneIntoDir, 0755); err != nil {
+		fail("Can't create directory (%s), error: %v", configs.CloneIntoDir, err)
+	}
 
-		if commitHash, err := git.LogCommitHash(); err != nil {
-			log.Errorf("Git log failed, error: %s", err)
-			os.Exit(1)
-		} else {
-			log.Printf("=> GIT_CLONE_COMMIT_HASH")
-			log.Printf("   value: %s", commitHash)
-			fmt.Println()
+	if err := run(Git.Init()); err != nil {
```

Comment:
would not it fail if repo already initialized? (like in case: `originPresent && configs.ResetRepository != "yes"`)